src/v/cluster/types.h

```diff
@@ -404,6 +404,7 @@ struct topic_properties_update {
 // Structure holding topic configuration, optionals will be replaced by broker
 // defaults
 struct topic_configuration {
+    topic_configuration() = default;
```

Comment:
i don't think this default ctor makes sense.

Reply:
got rid of it